internal/delta_map_downloader.go

```diff
@@ -0,0 +1,108 @@
+package internal
+
+import (
+	"github.com/pkg/errors"
+	"github.com/tinsane/storages/storage"
+	"github.com/wal-g/wal-g/internal/walparser"
+)
+
+func GetDeltaMap(folder storage.Folder, timeline uint32, firstUsedLSN, firstNotUsedLSN uint64) (PagedFileDeltaMap, error) {
+	deltaMap := NewPagedFileDeltaMap()
+	firstUsedDeltaNo, firstNotUsedDeltaNo := GetDeltaRange(firstUsedLSN, firstNotUsedLSN)
+	// Get locations from [firstUsedDeltaNo, lastUsedDeltaNo). We use lastUsedDeltaNo in next step
+	locationsFromDeltas, err := GetLocationsFromDeltas(folder, timeline, firstUsedDeltaNo, firstNotUsedDeltaNo.Previous())
```

Comment:
Amount of memory used here can exceed RAM size. Delta map should be built online.